libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentParser.java

```diff
@@ -117,6 +117,10 @@ public boolean isValue() {
         INT, LONG, FLOAT, DOUBLE
     }
 
+    interface MapValueParser<T> {
```

Comment:
We do have `CheckedFunction` which you could use instead of adding a new interface?

Reply:
if it looks terrible down the road then rather keep it but it should be a `@FunctionalInterface` IMO